library/src/main/java/com/pokegoapi/api/gym/Gym.java

```diff
@@ -161,6 +166,50 @@ private GymGetInfoResponse details() throws RequestFailedException {
 		return data;
 	}
 
+	private int getStartQuantity(Item item) throws RequestFailedException {				
+		return getApi().getInventories().getItemBag().getItem(item.getItemId()).getCount();		
+	}
+	
+	private GymFeedPokemonResponse getFeedPokemon(Item item, MotivatedPokemon motivatedPokemon) 
+			throws RequestFailedException {
+		List<TutorialState> tutorialStates = getApi().getPlayerProfile().getTutorialState().getTutorialStates();
+		if (!tutorialStates.contains(TutorialState.GYM_TUTORIAL)) {
+			getApi().getPlayerProfile().visitGymComplete();
+		}
+		int startingQuantity = getStartQuantity(item);
+		if (startingQuantity > 0) {
+			GymFeedPokemonMessage reqMsg = GymFeedPokemonMessage
+					.newBuilder()
+					.setGymId(this.getId())
+					.setItem(item.getItemId())
+					.setPlayerLatDegrees(getApi().getLatitude())
+					.setPlayerLngDegrees(getApi().getLongitude())
+					.setPokemonId(motivatedPokemon.getPokemon().getId())
+					.setStartingQuantity(startingQuantity)				
+					.build();
+	
+			ServerRequest serverRequest = new ServerRequest(RequestType.GYM_FEED_POKEMON, reqMsg);
+			getApi().getRequestHandler().sendServerRequests(serverRequest, true);
+	
+			try {
+				feedPokemon = GymFeedPokemonResponse.parseFrom(serverRequest.getData());
+				if (feedPokemon.getResult().equals(Result.SUCCESS)) {
+					item.setCount(startingQuantity - 1);
+				}
+			} catch (InvalidProtocolBufferException e) {
+				throw new RequestFailedException();
+			}
+		} else {
+			throw new RequestFailedException("Not Enough Item");
```

Comment:
It may be better to return a response with result `ERROR_NO_BERRIES_LEFT`. Maybe also handle other cases like being on the incorrect team.